daemon/cmd/daemon_main.go

```diff
@@ -1078,6 +1078,9 @@ func initializeFlags() {
 	flags.StringSlice(option.VtepMAC, []string{}, "List of VTEP MAC addresses for forwarding traffic outside the cluster")
 	option.BindEnv(option.VtepMAC)
 
+	flags.Int(option.TCFilterPriority, 1, "Priority of TC BPF filter")
+	option.BindEnv(option.TCFilterPriority)
```

Comment:
I'd probably rather make this a hidden option and not even expose it to the agent as an official flag. Reason is that if Cilium is not the first program in the tc chain, then there is no guarantee that it can function properly since it heavily depends on what the first program is doing and whether it is terminating the tc pipeline early or not. If the former, then it would bypass all of Cilium. (cc @joestringer) What do you plan it to chain it with?

Reply:
We want to be able to set the skb priority to redirect to a specific queue group.
I understand your concern and basically agree. Does making the option "hidden" restrict sharing it with users?

Reply:
> Does making the option "hidden" restrict sharing it with users?

No, i just means it's not exposed via --help. You can use it normally otherwise, and the Helm `extraArgs` to configure this specific scenario as @joestringer mentioned.

(Btw, you can also set skb->priority out of BPF as you probably know anyway. Independent of this PR, but maybe it could make sense to integrate something generically applicable also directly into Cilium?)

Reply:
@intel-dlanders I'm holding my review until this conversation is resolved as it will impact the PR's Helm changes.

Reply:
Well, I've made the argument hidden but I'm having a difficult time getting it into the agent daemonset yaml. I finally was able to use '--set extraArgs={--tc-filter-priority=6}' on the helm command line to add this arg but only after modifying the daemonset yaml file by inserting '- ' in front of the line '{{- toYaml . | trim | nindent 8 }}'. Then, and only then, I can pass '--set "extraArgs={--tc-filter-priority=6,--second-param=two}" and get the following in the config yaml generated by Helm:

```
args:
- --config-dir=/tmp/cilium/config-map
- --tc-filter-priority=6
- --second-param=two
```

Am I doing something wrong or is the extraArgs template not working correctly? Thanks

Reply:
@joestringer Can you help me with the extraArgs issue above? Thanks

Reply:
@intel-dlanders I'm not sure I understand, what's the problem? It seems to render out fine for me:

```
$ helm template cilium ./install/kubernetes/cilium --set extraArgs="{--foo,--bar}" --debug | grep -C 1 foo
install.go:178: [debug] Original chart version: ""
install.go:199: [debug] CHART PATH: /home/joe/git/cilium/install/kubernetes/cilium

        - --config-dir=/tmp/cilium/config-map
        - --foo
        - --bar
```